src/Core/Types/PrimitiveType.cs

```diff
@@ -158,6 +158,28 @@ public static PrimitiveType CreateWord(int bitSize)
 			return Create(dom, (short) bitSize, name);
 		}
 
+        //$TODO: Remove when PrimitiveType.CreateWord will accept a bit count.
+        public static PrimitiveType CreateWordFromBits(uint bitSize)
```

Comment:
Shouldn't be removed?

Reply:
Let's have both a `uint` and an `int` version of `PrimitiveType.CreateWord` for convenience. C# is stricter than C / C++ when it comes to unsignedness and all those casts are a pain.

Reply:
OK, shall do in next PR.

Reply:
What shall we do, if by chance, int bitSize is <= 0? Negate? Make it bool (forcing to 1)?

Reply:
Throw `ArgumentException` for negative sizes

Reply:
Or as you say ArgumentOutOfRangeException

Reply:
In fact, throw `ArgumentOutOfRangeException` for `bitSize <= 0` in `CreateWord(int bitSize)`.

Reply:
This is what I'll implement. However, I'm getting regression using those CreateWord() due to the fact there is no corresponding Constant. (ex. I'm having 24-bit type for some PIC registers but I can't defined 24-bit Constant by using current Constant.Create.